src/robot/api/deco.py

```diff
@@ -14,9 +14,19 @@
 #  limitations under the License.
 
 import inspect
+from typing import Any, Callable, Sequence, Type, TypeVar, overload
 
+from robot.api.interfaces import TypeHints
 
-def not_keyword(func):
+
+# Current annotations report `attr-defined` errors. This can be solved once Python 3.10
+# becomes the minimum version (error-free conditional typing proved too complex).
+# See: https://discuss.python.org/t/questions-related-to-typing-overload-style/38130
+F = TypeVar("F", bound=Callable[..., Any])
+Converters = dict[type, Callable[[Any], Any]]
```

Comment:
Converter can actually also accept two arguments. I don't know is there any nicer syntax than `Callable[[Any], Any] | Callable[[Any, Any], Any]` for that.